spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+            // Create client.
+            DatabaseAdminClient databaseAdminClient =
+                await DatabaseAdminClient.CreateAsync();
+            // Initialize request argument(s).
+            InstanceName parent = new InstanceName(projectId, instanceId);
+            string createStatement =
+                $"CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle) "
+                + "STORING (MarketingBudget)";
+            // Make the request.
+            var createDatabaseRequest =
+                await databaseAdminClient.UpdateDatabaseDdlAsync(
+                    new DatabaseName(projectId, instanceId, databaseId),
+                new[] { createStatement });
+            // Poll until the returned long-running operation is complete.
+            var updateDatabaseResponse =
+                await createDatabaseRequest.PollUntilCompletedAsync();
+            return updateDatabaseResponse.Result;
+            // [END create_storing_index]
+        }
+
+        public static async Task<object> AddColumnAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START add_column]
+            // Create client.
+            DatabaseAdminClient databaseAdminClient =
+                await DatabaseAdminClient.CreateAsync();
+            // Initialize request argument(s).
+            InstanceName parent = new InstanceName(projectId, instanceId);
+            string alterStatement =
+                $"ALTER TABLE Albums ADD COLUMN MarketingBudget INT64";
+            // Make the request.
+            var createDatabaseRequest =
+                await databaseAdminClient.UpdateDatabaseDdlAsync(
+                    new DatabaseName(projectId, instanceId, databaseId),
+                new[] { alterStatement });
+            // Poll until the returned long-running operation is complete.
+            var updateDatabaseResponse =
+                await createDatabaseRequest.PollUntilCompletedAsync();
+            Console.WriteLine("Added the MarketingBudget column.");
+            return updateDatabaseResponse.Result;
+            // [END add_column]
+        }
+
+        public static async Task QuerySampleDataAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/"
+            + $"{instanceId}/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId : {reader["SingerId"]} "

```

Comment:
I don't think spanner would allow you to change FLOAT64 to INT64 or vice versa. So in practice this might not matter. I think the only type transformation we allow is bytes to string but I am not sure.